apps/omg_child_chain/lib/omg_child_chain/fee_server.ex

```diff
@@ -95,34 +87,42 @@ defmodule OMG.ChildChain.FeeServer do
     new_state =
       case update_fee_specs(state) do
         {:ok, updated_state} ->
-          Alarm.clear(Alarm.invalid_fee_file(__MODULE__))
+          Alarm.clear(Alarm.invalid_fee_source(__MODULE__))
           updated_state
 
         :ok ->
-          Alarm.clear(Alarm.invalid_fee_file(__MODULE__))
+          Alarm.clear(Alarm.invalid_fee_source(__MODULE__))
           state
 
         _ ->
-          Alarm.set(Alarm.invalid_fee_file(__MODULE__))
+          Alarm.set(Alarm.invalid_fee_source(__MODULE__))
           state
       end
 
     {:noreply, new_state}
   end
 
   @spec update_fee_specs(map()) :: :ok | {:ok, map()} | {:error, list({:error, atom(), any(), non_neg_integer() | nil})}
-  defp update_fee_specs(%{fee_adapter: adapter, expire_fee_timer: current_timer} = state) do
+  defp update_fee_specs(
```

Comment:
What other options we have? I could move this pattern match below to the function body...
but right now in the other places we tend to "unstructure" in function parameters

Reply:
if you follow the earlier comment I made about using a struct as the state structure, a completely reasonable approach is to do 
```
state.fee_adapter
state.fee_adapter_opts
state.expire_fee_timer
state.fee_buffer_duration_ms
```

Reply:
I'd like to disagree here - also in fact - we were instructed to destructure multiple vars in function header (I can recall the CR's comment I was given)

Reply:
at line https://github.com/omisego/elixir-omg/pull/1373/files#diff-a3c546fa6ad28e56859c0ca99ddd9729R61
you've used this pattern, I don't quite get why not in update_fee_specs.

Reply:
TL;DR won't change, I prefer not to do 180 from the previous experience

I've scrolled many `Core` modules where we most likely consume structs as a function params. It seems that we tend to pattern match where there is 2 or more fields of the struct used in the function body.
Our style guide does not impose it, also I can recall review comment asking me to prefer pattern-match in such situation as more readable approach.
I'd like to stay consistent with which seems to me the most common code pattern.

Reply:
Some references
   - https://github.com/omisego/elixir-omg/blob/ec998841f95b261f7077c936f7d474b5017dbe20/apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex#L418
   - https://github.com/omisego/elixir-omg/blob/ec998841f95b261f7077c936f7d474b5017dbe20/apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex#L449
   - https://github.com/omisego/elixir-omg/blob/ec998841f95b261f7077c936f7d474b5017dbe20/apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex#L493
   - https://github.com/omisego/elixir-omg/blob/af232fce103f42dd6b960433a9d406b67d22df54/apps/omg/lib/omg/state/core.ex#L249
   - https://github.com/omisego/elixir-omg/blob/af232fce103f42dd6b960433a9d406b67d22df54/apps/omg/lib/omg/state/core.ex#L273
   - https://github.com/omisego/elixir-omg/blob/af232fce103f42dd6b960433a9d406b67d22df54/apps/omg/lib/omg/state/core.ex#L432
   - https://github.com/omisego/elixir-omg/blob/5b7e3db2f45e2a68169b94e565683a444acdc174/apps/omg/lib/omg/root_chain_coordinator/core.ex#L114
   - https://github.com/omisego/elixir-omg/blob/5b7e3db2f45e2a68169b94e565683a444acdc174/apps/omg/lib/omg/root_chain_coordinator/core.ex#L182
   - https://github.com/omisego/elixir-omg/blob/db3349a5648f05f370976d57b72ffbd39fb47d2d/apps/omg_watcher/lib/omg_watcher/block_getter/core.ex#L247
   - https://github.com/omisego/elixir-omg/blob/db3349a5648f05f370976d57b72ffbd39fb47d2d/apps/omg_watcher/lib/omg_watcher/block_getter/core.ex#L371
   - https://github.com/omisego/elixir-omg/blob/d85b93c1e4928815ae4e6274c062c4a7634b2ac7/apps/omg/lib/omg/ethereum_event_listener/core.ex#L100
   - https://github.com/omisego/elixir-omg/blob/44fe21b745723fffd7451b3fd8b7f0289da8a0c4/apps/omg_watcher/lib/omg_watcher/exit_processor/core.ex#L164
   - https://github.com/omisego/elixir-omg/blob/44fe21b745723fffd7451b3fd8b7f0289da8a0c4/apps/omg_watcher/lib/omg_watcher/exit_processor/core.ex#L395
   - https://github.com/omisego/elixir-omg/blob/44fe21b745723fffd7451b3fd8b7f0289da8a0c4/apps/omg_watcher/lib/omg_watcher/exit_processor/core.ex#L443

Reply:
old practices have shown not be the most effective approach in terms of readability. all the code above you're pointing to is code that we're trying to rewrite/optimise (not performance, but readability).

Reply:
ask @achiurizo if he likes how core modules look like :)

Reply:
> we were instructed to destructure multiple vars in function header (I can recall the CR's comment I was given)

Where's the comment? Interested in seeing the perspective there.

I think this case, it would be much more readable if we considered using a struct accessor as @InoMurko mentioned and that you had done earlier in the code. I myself find it hard to determine when's a good time to have long pattern matches like this, but it makes sense that:

* this is within the context of the structure, so it's clear this function is operating on that structure
* this is a private function, so it could be assume we can expect what the data looks like as we are passing it internally ourselves in this context.

> I'd like to stay consistent with which seems to me the most common code pattern.

While this common code pattern is common for this repo, and I don't know if that's a particularly good strategy if we want to continue to refactor and improve the code-base. This can be said about most other code bases too.

What could help is if we come to a consensus here on how we make determinations when we want to do exhaustive pattern matches like this or us something like the struct accesor and assume a context like this, the function is an extension of that data structure.

> ask @achiurizo if he likes how core modules look like :)

I am not a fan.

Reply:
![image](https://user-images.githubusercontent.com/2582555/77699186-3444b000-6fb2-11ea-9268-ebc475fb4b93.png)

Reply:
Thanks for the comment @achiurizo , @InoMurko 

The previous CR comment I mentioned: I’m unable to find it unfortunately. Probably nothing worth the effort.

Internet doesn’t seem to have strong opinion I just find [interesting comment](https://gist.github.com/madlep/d102f5cb1d578064d0c4#gistcomment-1623048) about compile time check.

Personally pattern-match feels more elixir style to me, but the consistency is the argument here. If we decide to change it to the struct accessor we should address it in a PR and reflect this in style guide.
Judging on subjective readability taste isn’t  always a good premise.